src/amo/components/Search/index.js

```diff
@@ -60,49 +77,66 @@ export class SearchBase extends React.Component {
       LinkComponent,
       count,
       enableSearchSort,
+      errorHandler,
       filters,
-      hasSearchParams,
       loading,
-      page,
+      paginationQueryParams,
```

Comment:
The issue with that is it needlessly pollutes the query params for things like category pages, because filters like `addonType` and `category` already exist elsewhere in the URL. Without explicitly specifying `paginationQueryParams` category navigating from the first to second page does this:

```
# First page's URL
/en-GB/firefox/feeds-news-blogging/
# Second page's URL
/en-US/firefox/extensions/feeds-news-blogging/?app=firefox&category=feeds-news-blogging&page=2&type=extension
```

The paginator will now (by default/with no `paginationQueryParams`) using the `filters` it's supplied to do exactly what you suggested, but this way we can make slightly nicer/less-repeated URLs.